protected/humhub/modules/content/components/ContentContainerActiveRecord.php

```diff
@@ -64,6 +66,26 @@ abstract class ContentContainerActiveRecord extends ActiveRecord
      */
     public $defaultRoute = '/';
 
+    /**
+     * @var array Related Tags which should be update after save
+     */
+    public $updatedTags;
+
+    /**
+     * @inheritdoc
+     */
+    public function __get($name)
+    {
+        switch ($name) {
```

Comment:
@buddh4 

> by default `$space->tags` will fall back to `$space->getTags()`

Are you sure?
If yes then why here https://github.com/humhub/humhub/pull/4867/files#diff-d3a4745ae7c92bb2ed3212a38c146f5f29268cf40a693d007f752f679a1f52abL696 was old code `($this->tags != '')` as comparing with string and not with array.
And old function `function getTags()` here https://github.com/humhub/humhub/pull/4867/files#diff-d3a4745ae7c92bb2ed3212a38c146f5f29268cf40a693d007f752f679a1f52abL704-L707 returned array.
Also in old version we had `$form->field($model, 'tags')` and the `$space->tags` couldn't fallback to `$space->getTags()` because otherwise we got error there but it was printed as simple text input with string value like `tag1, tag 2, tag N`.

So I specially implemented the case for old property `tags` in `function __get($name)` in order to keep it as it was before, i.e `$space->tags` still returns a string and `$space->getTags()` - array as it was before this PR.

I decided to keep this in order to don't break external modules like RestAPI https://github.com/humhub/humhub-modules-rest/blob/master/definitions/UserDefinitions.php#L60, https://github.com/humhub/humhub-modules-rest/blob/master/definitions/SpaceDefinitions.php#L43 and maybe some other modules.

Please correct me if I am wrong somewhere.

Reply:
See https://www.yiiframework.com/doc/guide/2.0/en/db-active-record#accessing-relational-data

Reply:
@buddh4 I have read the doc pages, there is an example:

 - `$customer->orders;` // is an **array** of `Order` objects
 - `$customer->getOrders();` // returns an **ActiveQuery** instance

I.e. the `$model->tags` should NOT return the same as `$model->getTags()`.
However yes partly you are right to keep the behaviour we need:

- `$model->tags` - returns an **array** of Tags objects
- `$model->getTags()` - returns an **ActiveQuery** instance

> What is the reason you overwrite the magic getter, is this required in order to stay compatible?

Yes, as I wrote above I did this in order to return same results as it was before implementing tags in separate table, because some external(not core) modules use `$user->tags` and `$space->tags`.

In previous and in current versions:

- `$model->tags` - returns a **string** of tags separated by `, `
- `$model->getTags()` - returns an **array** of tags strings

Do you think we should completely change the return result of the property and method to keep the behaviour as the doc pages? and then create PR for all external modules where the `$model->tags` and `$model->getTags()` are used.

Reply:
@yurabakhtin yes sorry In case of returning an ActiveQuery it does not return the same as the getter but instead the cached result, but for normal non ActiveQuery returns it should return the same result. If this behavior did not change by this PR and its noted in the PHPDoc, I think its ok.

Generally I just think we hardly ever need a comma seperated string of those tags, since when processing it we need to split it anyways in most or all use-cases. Did you identify where this `$model->tags` is used?

Reply:
It would be fine for me if we refactor(remove) all `$space|$user->tags` uses and always switch to `->getTags()`.
For transition (e.g. until 1.10)  with custom themes/modules, we can keep the `__get` way and drop  a warning if used?

Reply:
@buddh4 

> Did you identify where this `$model->tags` is used?

- https://github.com/humhub/humhub-modules-rest/blob/master/definitions/UserDefinitions.php#L60
- https://github.com/humhub/humhub-modules-rest/blob/master/definitions/SpaceDefinitions.php#L43
- https://github.com/humhub/humhub/blob/master/protected/humhub/modules/space/widgets/Chooser.php#L26
- https://git.humhub.org/professional/enterprise-theme/-/blob/master/controllers/BrowseController.php#L48

Let me know if I should modify them all even and create PR in external modules.